bench/wtperf/wtperf.c

```diff
@@ -755,19 +770,168 @@ worker(void *arg)
 				strncpy(value_buf,
 				    value, opts->value_sz_max - 1);
 				if (workload->update_delta != 0)
-					update_value_delta(thread);
+					update_value_delta(thread,
+					    workload->update_delta);
 				if (value_buf[0] == 'a')
 					value_buf[0] = 'b';
 				else
 					value_buf[0] = 'a';
 				if (opts->random_value)
-					randomize_value(thread, value_buf);
+					randomize_value(thread,
+					    value_buf,
+					    workload->update_delta);
+
 				cursor->set_value(cursor, value_buf);
 				if ((ret = cursor->update(cursor)) == 0)
 					break;
 				goto op_err;
 			}
 
+			/*
+			 * Reads can fail with WT_NOTFOUND: we may be searching
+			 * in a random range, or an insert thread might have
+			 * updated the last record in the table but not yet
+			 * finished the actual insert.  Count failed search in
+			 * a random range as a "read".
+			 */
+			if (ret == WT_NOTFOUND)
+				break;
+			goto op_err;
+		case WORKER_MODIFY:
+			if ((ret = cursor->search(cursor)) == 0) {
+				if ((ret = cursor->get_value(
+				    cursor, &value)) != 0) {
+					lprintf(wtperf, ret, 0,
+					    "get_value in update.");
+					goto err;
+				}
+				/*
+				 * Copy as much of the previous value as is
+				 * safe, and be sure to NUL-terminate.
+				 */
+				strncpy(value_buf,
+				    value, opts->value_sz_max - 1);
+				if (workload->modify_delta != 0)
+					update_value_delta(thread,
+					    workload->modify_delta);
+				if (value_buf[0] == 'a')
+					value_buf[0] = 'b';
+				else
+					value_buf[0] = 'a';
+				if (opts->random_value)
+					randomize_value(thread,
+					    value_buf,
+					    workload->modify_delta);
+
+				/*
+				 * Distribute the modifications across whole
+				 * document.
+				 *
+				 * The distribution limits are same as MongoDB
+				 * that are imposed for an update operation to
+				 * be considered as modify or not?
+				 *
+				 * up to 10% of total modification size
+				 * 16 number of modifications per record
+				 *
+				 * NOTE: for records of size less than 160,
+				 * there is no distributed modifications.
+				 */
+
+				nmodifications = MAX_NUM_OF_MODIFICATIONS;
+
+				/* 10% of total size of the record */
+				total_modify_size = strlen(value) /
+				    MAX_SIZE_OF_MODIFICATION;
+
+				/*
+				 * Randomize the maximum modification size
+				 * and offset per modify. With randomization
```

Comment:
I think that only the first sentence of this comment is necessary.

Reply:
changed accordingly.